packages/stream_chat_flutter/lib/src/message_input/audio_message_send_button/stream_audio_message_send_button.dart

```diff
@@ -0,0 +1,224 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:stream_chat_flutter/src/message_input/audio_message_send_button/stream_audio_message_controllers.dart';
+import 'package:stream_chat_flutter/src/message_input/audio_message_send_button/stream_audio_message_gesture_state.dart';
+import 'package:stream_chat_flutter/src/message_input/audio_message_send_button/stream_audio_message_overlays.dart';
+import 'package:stream_chat_flutter/stream_chat_flutter.dart';
+
+/// A callback called when the user starts recording an audio message.
+typedef OnRecordingStart = void Function();
+
+/// A callback called when the user ends recording an audio message.
+typedef OnRecordingEnd = void Function();
+
+/// A callback called when the user cancels recording an audio message.
+typedef OnRecordingCanceled = void Function();
```

Comment:
Why not use `VoidCallback` here? (Not objecting to this, just curious) - since the user only sees the parameter name and using a custom callback is a bit confusing since they have no clue what parameters to use when passing the callback.

Reply:
I don't really like using other SDK's code as ours, even if it is from Flutter itself. Also this is provisioning for changes in the future 😉